// take-notes/dist/index.mjs
// src/students/Abhinav_K.ts
var Abhinav_K_Work = [
  {
    studentName: "Abhinav.K",
    gitHub: "ABHINAV-KEASAVAN",
    repo: "POC",
    fileName: "TEMPLATES AND IMAGE.pdf",
    language: "HTML",
    topic: "Tags in HTML"
  }
];

// src/students/Akash_B.ts
var Akash_BWork = [
  {
    studentName: "Akash B",
    gitHub: "Akashb0119",
    repo: "poc",
    fileName: "HTML Tags.pdf",
    language: "html",
    topic: "Brief Notes for HTML Tags"
  }
];

// src/students/Chandan.ts
var ChandanWork = [
  {
    studentName: "Chandan N",
    gitHub: "chandushaiva",
    repo: "Notes",
    fileName: "Final keyboard.pdf",
    language: "java",
    topic: "Final keyboard in java"
  },
  {
    studentName: "Chandan N",
    gitHub: "chandushaiva",
    repo: "Notes",
    fileName: "Abstract in java.pdf",
    language: "java",
    topic: "Abstract keyboard in java"
  },
  {
    studentName: "Chandan N",
    gitHub: "chandushaiva",
    repo: "Notes",
    fileName: "document     write.pdf ",
    language: "js",
    topic: "How to add image using document.write method"
  }
];

// src/students/Karthick_R.ts
var Karthick_R_Work = [
  {
    studentName: "Karthick R",
    gitHub: "karthickrajendran",
    repo: "Notes",
    fileName: "Javascript Notes.pdf ",
    language: "javascript",
    topic: "basics, \u2018variables and Data types\u2019, \u2018Functions\u2019, \u2018Conditional Statements\u2019, \u2018Loops\u2019, \u2018Arrays\u2019, \u2018Objects\u2019, \u2018DOM manipulation\u2019, \u2018Asynchronous\u2019, \u2018Prototypal inheritance\u2019, \u2018Closures\u2019, \u2018Promise\u2019, \u2018Higher order Functions\u2019, \u2018Modules\u2019, \u2018Callback and Error handling\u2019, \u2018Async/Await\u2019, \u2018Call stack\xA0and\xA0Event\xA0Loop\u2019"
  }
];

// src/students/Manoj.ts
var Manoj_Work = [
  {
    studentName: "Manoj M",
    gitHub: "skylinedude",
    repo: "My-notes",
    fileName: "eclipse_marketplace.pdf",
    language: "java",
    topic: "Eclipse Marketplace"
  },
  {
    studentName: "Manoj M",
    gitHub: "skylinedude",
    repo: "My-notes",
    fileName: "import_java_external_framework.pdf",
    language: "java",
    topic: "How to import external framework into Eclipse IDE"
  },
  {
    studentName: "Manoj M",
    gitHub: "skylinedude",
    repo: "My-notes",
    fileName: "mongodb-atlas-compass.pdf",
    language: "MongoDB",
    topic: "HOW TO CONNECT MONGODB ATLAS TO MONGODB COMPASS"
  }
];

// src/students/My_notes.ts
var waseem_Work = [
  {
    studentName: "Waseem Ahmed",
    gitHub: "wa5",
    repo: "all-topics-pdf",
    fileName: "Spring boot API Railway deployment.pdf",
    language: "Java deployment",
    topic: "Spring boot API Railway deployment"
  },
  {
    studentName: "Waseem Ahmed",
    gitHub: "wa5",
    repo: "all-topics-pdf",
    fileName: "diffrentBundler.pdf",
    language: "webpack",
    topic: "diffrent Bundler like webpack"
  },
  {
    studentName: "Waseem Ahmed",
    gitHub: "wa5",
    repo: "all-topics-pdf",
    fileName: "for-for-in-for-off.pdf",
    language: "js",
    topic: "for for-in for-off topics"
  },
  {
    studentName: "Waseem Ahmed",
    gitHub: "wa5",
    repo: "all-topics-pdf",
    fileName: "webpack-plugins-loaders.pdf",
    language: "webpack",
    topic: "webpack plugins loaders"
  },
  {
    studentName: "Waseem Ahmed",
    gitHub: "wa5",
    repo: "all-topics-pdf",
    fileName: "webpack.pdf",
    language: "webpack",
    topic: "webpack "
  }
];

// src/students/Nataraj_m.ts
var Nataraj_M_Work = [
  {
    StudentName: "Nataraj M",
    gitHub: "nmahendran7",
    repo: "Insta",
    fileName: "How to Install PyCharm.pdf",
    language: "Python",
    topic: "Installation of\xA0PyCharm"
  },
  {
    StudentName: "Nataraj M",
    gitHub: "nmahendran7",
    repo: "Insta",
    fileName: "How to Install Python.pdf",
    language: "Python",
    topic: "Installation of Python"
  }
];

// src/students/Prajwal Raj N.ts
var Prajwal_Raj_NWork = [
  {
    studentName: "Prajwal Raj N",
    gitHub: "prajwalrajn",
    repo: "My-notes",
    fileName: "How to Install Typescript.pdf",
    language: "javascript",
    topic: "How to Install Typescript"
  }
];

// src/students/Prathap_V.ts
var prathap_V_Work = [
  {
    studentName: " Prathap V",
    gitHub: "Prathap0703",
    repo: "poc",
    fileName: "HTML.docx",
    language: "HTML",
    topic: "Heading Tags"
  },
  {
    studentName: "Prathap V",
    gitHub: "Prathap0703",
    repo: "poc",
    fileName: "HTML Headings.pdf",
    language: "html",
    topic: "Heading\xA0Tags"
  }
];

// src/students/Ramkumar_S.ts
var Ramkumar_Work = [
  {
    studentName: "Ramkumar S",
    gitHub: "Ramkumar1121",
    repo: "POC",
    fileName: "INTRODUCTION OF HTML TAGS.pdf",
    language: "html",
    topic: "HTML Tags"
  }
];

// src/students/Rishabav_M.ts
var Rishabav_Work = [
  {
    StudentName: "Rishabav M",
    gitHub: "Rishabav-1",
    repo: "Installation",
    fileName: "Python Installation.pdf",
    language: "Python",
    topic: "Installation of Python"
  },
  {
    StudentName: "Rishabav M",
    gitHub: "Rishabav-1",
    repo: "Installation",
    fileName: "Pycharm Installation.pdf",
    language: "Python",
    topic: "Installation\xA0of\xA0Pycharm"
  }
];

// src/students/Safrin_Sara.ts
var Safrin_Sara_Work = [
  {
    studentName: "Safrin Sara",
    gitHub: "codehero11",
    repo: "new-project",
    fileName: " wireframe.pdf",
    language: "java",
    topic: "wireframe"
  }
];

// src/students/Saima_Nasim.ts
var Saima_Nasim_Work = [
  {
    studentName: "Saima Nasim",
    gitHub: "saimanasim",
    repo: "practice",
    fileName: "Saima_Twilio.pdf",
    language: "js",
    topic: "Twilio introduction and how to send SMS in Twilio"
  },
  {
    studentName: "Saima Nasim",
    gitHub: "saimanasim",
    repo: "practice",
    fileName: "github_hosting.pdf",
    language: "github",
    topic: "How to host github repository"
  }
];

// src/students/Venkatesh C K.ts
var Venkatesh_C_KWork = [
  {
    studentName: "Venkatesh C K",
    gitHub: "VenkateshCK071",
    repo: "My-notes",
    fileName: "command.pdf",
    language: "ts",
    topic: "How to execute TypeScript file using command line"
  }
];

// src/students/Wajhuddin_gauri.ts
var Wajhuddin_gauri_Work = [
  {
    studentName: "Wajhuddin Gauri",
    gitHub: "mugalnepal",
    repo: "practice",
    fileName: "Sending-mail.pdf",
    language: "js",
    topic: "sending email using javascript"
  }
];

// src/students/shalini.ts
var shaliniWork = [
  {
    studentName: "Shalini S",
    gitHub: "Shalini1499",
    repo: "POC",
    fileName: "html.pdf",
    language: "Html",
    topic: "Html Tags"
  }
];

// src/data.ts
var data = [
  {
    students: [
      {
        studentName: "jayanth",
        gitHub: "BR-Jayanth",
        repo: "take-notes-lib",
        fileName: "Jayanthbr.pdf",
        language: "js",
        topic: "for loop"
      },
      {
        studentName: "Safrin Sara",
        gitHub: "codehero11",
        repo: "new-project",
        fileName: "JAVA.pdf",
        language: "java",
        topic: "int"
      },
      {
        studentName: "Safrin Sara",
        gitHub: "codehero11",
        repo: "new-project",
        fileName: " hashmap and hashset.pdf",
        language: "java",
        topic: "hashmap and hashset"
      },
      {
        studentName: "Prajwal Raj N",
        gitHub: "prajwalrajn",
        repo: "My-notes",
        fileName: "Integerjava.pdf",
        language: "java",
        topic: "Integer"
      },
      {
        studentName: "Prajwal Raj N",
        gitHub: "prajwalrajn",
        repo: "My-notes",
        fileName: "arraylist and linkedlist.pdf",
        language: "java",
        topic: "ArraylList and LinkedList"
      },
      {
        studentName: "Safrin Sara",
        gitHub: "codehero11",
        repo: "new-project",
        fileName: " array and linked list.pdf",
        language: "java",
        topic: " array and linked list"
      },
      {
        studentName: "Safrin Sara",
        gitHub: "codehero11",
        repo: "new-project",
        fileName: " java char.pdf",
        language: "java",
        topic: "char"
      },
      {
        studentName: "Manoj M",
        gitHub: "skylinedude",
        repo: "My-notes",
        fileName: "Hashmap.pdf",
        language: "java",
        topic: "Diffrence between HashMap and Hashset"
      },
      {
        studentName: "Manoj M",
        gitHub: "skylinedude",
        repo: "My-notes",
        fileName: "arraylist.pdf",
        language: "java",
        topic: "Diffrence between ArrayList and LinkedList"
      },
      {
        studentName: "Manoj M",
        gitHub: "skylinedude",
        repo: "My-notes",
        fileName: "Java Strings.pdf",
        language: "java",
        topic: "Strings"
      },
      {
        studentName: "Chandan N",
        gitHub: "skylinedude",
        repo: "My-notes",
        fileName: "Abstract in java.pdf",
        language: "java",
        topic: "Abstract Keyword"
      },
      {
        studentName: "Jyothika D",
        gitHub: "jeeeeeva",
        repo: "take-notes-lib",
        fileName: "Data_Types in Javascript.pdf",
        language: "js",
        topic: "Datatypes in javascript"
      },
      {
        studentName: "Sudeep Raj",
        gitHub: "sudeeprajbs",
        repo: "take-notes-lib",
        fileName: "hoisting_by_sudeep.pdf",
        language: "js",
        topic: "Hoisting"
      },
      {
        studentName: "Rakesh D K",
        gitHub: "Rocky-1999",
        repo: "take-notes-lib",
        fileName: "hosting.pdf",
        language: "js",
        topic: "Hoisting"
      },
      {
        studentName: "pujitha",
        gitHub: "1pujitha",
        repo: "take-notes-lib",
        fileName: "pujitha-pdf.pdf",
        language: "js",
        topic: "const"
      },
      {
        studentName: "Prathap.K",
        gitHub: "Iam-Prathap",
        repo: "take-notes-lib",
        fileName: "JAVASCRIPT-STRINGS.pdf",
        language: "js",
        topic: "Strings"
      },
      {
        studentName: "Sudarshana B U",
        gitHub: "Sudarshana-17",
        repo: "take-notes-lib",
        fileName: "Let-Keyword.pdf",
        language: "js",
        topic: "Let keyword"
      },
      ...shaliniWork,
      ...Akash_BWork,
      ...prathap_V_Work,
      ...Abhinav_K_Work,
      ...Manoj_Work,
      ...Safrin_Sara_Work,
      ...Saima_Nasim_Work,
      ...Ramkumar_Work,
      ...Wajhuddin_gauri_Work,
      ...Venkatesh_C_KWork,
      ...Prajwal_Raj_NWork,
      ...ChandanWork,
      ...Nataraj_M_Work,
      ...Rishabav_Work,
      ...Karthick_R_Work,
      ...waseem_Work
    ]
  }
];

// src/allNotes.ts
var base = `https://cdn.jsdelivr.net/gh`;
var urlBulder = `${base}`;
var allNotes = {
  all: () => {
    let added = data[0].students.map((val) => {
      let base2 = val;
      let exurlBulder = `${urlBulder}/${base2.gitHub}/${base2.repo}/${base2.fileName}`;
      return { ...base2, fileUrl: exurlBulder };
    });
    return added;
  }
};
var students = {
  findOne: (name) => {
    let info = data[0].students.find((val) => {
      return val.studentName === name;
    });
    let exurlBulder = `${urlBulder}/${info?.gitHub}/${info?.repo}/${info?.fileName}`;
    return { fileurl: exurlBulder, studentInfo: info };
  },
  findAll: (name) => {
    let info = data[0].students.filter((val) => {
      return val.studentName === name;
    });
    let added = info.map((val) => {
      let exurlBulder = `${urlBulder}/${val?.gitHub}/${val?.repo}/${val?.fileName}`;
      return { ...val, fileUrl: exurlBulder };
    });
    return added;
  }
};
export {
  allNotes,
  students
};
